mypy/fastparse.py

```diff
@@ -213,10 +214,64 @@ def parse(
         assert options.python_version[0] >= 3
         feature_version = options.python_version[1]
     try:
+        ignore_comment_errors = options and options.ignore_comment_errors
+
+        def try_parse_ast(source: str | bytes) -> tuple[AST | None, str | bytes]:
+            try:
+                ast = ast3_parse(source, fnam, "exec", feature_version=feature_version)
+            except SyntaxError as e:
+
+                def strip_comment(line: str) -> str:
+                    return re.sub(r"([^#]*).*(\r\n|\r|\n)?", r"\1\2", line)
+
+                def strip_commentb(line: bytes) -> bytes:
+                    return re.sub(rb"([^#]*).*(\r\n|\r|\n)?", rb"\1\2", line)
+
+                def strip_comment_from_line(text: str | bytes, target_lineno: int) -> str | bytes:
+                    if isinstance(text, str):
+                        return "".join(
+                            line if lineno != target_lineno else strip_comment(line)
+                            for lineno, line in enumerate(text.splitlines(keepends=True), start=1)
+                        )
+                    return b"".join(
+                        line if lineno != target_lineno else strip_commentb(line)
+                        for lineno, line in enumerate(text.splitlines(keepends=True), start=1)
+                    )
+
+                if (
+                    ignore_comment_errors
+                    and e.text
+                    and e.lineno
+                    and TYPE_COMMENT_PATTERN.match(e.text)
+                ):
+                    stripped_source = strip_comment_from_line(source, e.lineno)
+                    if errors:
+                        errors.report(
+                            e.lineno or -1,
+                            e.offset or -1,
+                            # TODO: It would be nicer if the message included the line text via
+                            # `e.text.strip()`. However, I can't figure out how to make the
+                            # unit test pass, because the # N: <expected_result> in the unit
+                            # test must contain itself. If there was some partial matching
+                            # mechanism for the unit test comment, then we could make this work.
```

Comment:
To resurface previous comment (the thread was made 'obsolete' by a push), you can list the expected errors in a separate `[out]` section if the inline assertions are harder in a particular case.

But also I'd rather have just the `# type: ...` comment, e.g.
```
Ignored bad type comment ("# type: bad")
```
For this, I'd probably split the line over the `# type:` -- maybe use TYPE_COMMENT_PATTERN (modify it to capture both before and after the # type) to match and then
- use the 1st group as the replacement line
- use the 2nd group in the error message

Reply:
Cool. I found `out` in the doc. It was 2 lines below where I was reading about `E` and `N`. Talk about tunnel vision! I'll work on this.